request.go

```diff
@@ -0,0 +1,110 @@
+package typhon
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/mondough/slog"
+	"github.com/mondough/terrors"
+	"golang.org/x/net/context"
+)
+
+type Request struct {
+	http.Request
+	context.Context
+}
+
+// Encode serialises the passed object as JSON into the body (and sets appropriate headers).
+func (r *Request) Encode(v interface{}) {
```

Comment:
I guess at some point it would be nice to add support for alternative encoders, but we can still use this interface at that point anyway 😄 

The `Decode` method below returns an `error`, should this not too? Otherwise we presumably can only determine encoding failures from logs, and possibly weird errors as the `Content-Type` header & body not being set. I'm guessing that the `Request` type not having an `Error` param (unlike `Response`) and `NewRequest` not returning an error makes this more difficult, hence [this](https://github.com/mondough/typhon/blob/5facb016801ef1d2cf81dbcac5cdb44f703b6bef/request.go#L101), but given it would be a single-value return we could still return the error here?